src/nvim/buffer_defs.h

```diff
@@ -814,6 +814,8 @@ struct tabpage_S {
   frame_T         *(tp_snapshot[SNAP_COUNT]);    /* window layout snapshots */
   dictitem_T tp_winvar;             /* variable for "t:" Dictionary */
   dict_T          *tp_vars;         /* internal variables, local to tab page */
+  char_u          *tp_localdir;     /* absolute path of local directory or
+                                       NULL */
```

Comment:
The struct has not been refactored, that's why I chose to stick with the old ways. Mixing new and old style in an old style struct seemed like a bad idea to me. That's why I also added the `tp_` prefix to the member variable.